packages/graphql-hooks/src/useQuery.js

```diff
@@ -46,20 +45,57 @@ function useQuery(query, opts = {}) {
     }
   }, [state.error, allOpts.throwErrors])
 
+  const refetch = React.useCallback(
+    (options = {}) =>
+      queryReq({
+        skipCache: true,
+        // don't call the updateData that has been passed into useQuery here
+        // reset to the default behaviour of returning the raw query result
+        // this can be overridden in refetch options
+        updateData: (_, data) => data,
+        ...options
+      }),
+    [queryReq]
+  )
+
+  React.useEffect(
+    function subscribeToMutationsAndRefetch() {
+      if (!Array.isArray(opts.refetchAfterMutations)) return
+
+      const mutationsMap = opts.refetchAfterMutations.reduce((acc, value) => {
+        // value comes from useClientRequest and contains the mutation result
+        // in case that we want to do something more with it
+        acc[value.mutation] = {
+          filter: value.filter
+        }
+
+        return acc
+      }, {})
+
+      const conditionalRefetch = ({ mutation, variables }) => {
+        const { filter } = mutationsMap[mutation]
+
+        if (!filter || (variables && filter(variables))) {
+          refetch()
```

Comment:
how do we handle the case when a component unmounts in the middle of a refetch? won't we have to handle the error here or is it handled already by this library somewhere else? iow it's the typical problem that you trigger an async operation which will need to update local state, and while the operation is in progress the component unmounts. if you try to update the state of an unmounted component you'll get a React error. 
here we don't have access to the state because it's stored elsewhere, but I wonder if we need to catch and swallow the error instead

Reply:
This makes me think about this change in React:
* https://github.com/facebook/react/pull/22114

I need to think if our case is different and needs special adjustments, but it might not.

Reply:
Ah very interesting, I wasn't aware of this! I see it's merged, do we know if it's released too? I'm fairly sure I've seen this error still occurring recently

Reply:
It is not released yet. The PR is from August while the last release of React was in March.
* I think that we shouldn't manage this situation though and treat the error message as an expected false positive
* the message is useful in case we forget to clean up the subscription (but we didn't forget about that)
* React is using `console.error` and not throwing an Error, so it wouldn't help to "catch"

Please let me know if you have different thoughts about this.

Reply:
sounds perfectly reasonable